src/client/datascience/kernel-launcher/kernelProcess.ts

```diff
@@ -160,7 +160,7 @@ export class KernelProcess implements IKernelProcess {
                         });
                         exitEventFired = true;
                     }
-                    cancelWaiting.cancel();
+                    deferred.reject(error);
```

Comment:
Better than cancellation, to preserve the original error, else we loose a lot of it when its bubbled up.